File: services/common/src/main/java/com/example/common/validator/IntegerValidator.java
package com.example.common.validator;

public class IntegerValidator implements Validator {
    private final Integer integer;
    private boolean result;

    public IntegerValidator(Integer integer) {
        this.result = true;
        this.integer = integer;
    }

    public IntegerValidator notNull() {
        this.result &= this.integer != null;
        return this;
    }

    public IntegerValidator greaterThanZero() {
        this.result &= this.integer > 0;
        return this;
    }

    @Override
    public boolean isValid() {
        return this.result;
    }
}

File: services/common/src/main/kotlin/com/example/common/validator/IntegerValidator.kt
package com.example.common.validator

class IntegerValidator(private val integer: Int): Validator {
    private var result = true

    fun greaterThanZero(): IntegerValidator {
        result = result and (integer > 0)
        return this
    }

    override fun isValid(): Boolean {
        return result
    }
}